src/Microsoft.Extensions.Configuration.AzureAppConfiguration/JsonKeyValueAdapter.cs

```diff
@@ -30,18 +30,23 @@ internal class JsonKeyValueAdapter : IKeyValueAdapter
             }
 
             string rootJson = $"{{\"{setting.Key}\":{setting.Value}}}";
-            JsonElement jsonData;
+
+            List<KeyValuePair<string, string>> keyValuePairs = new List<KeyValuePair<string, string>>();
+
             try
             {
-                jsonData = JsonSerializer.Deserialize<JsonElement>(rootJson);
+                using (JsonDocument document = JsonDocument.Parse(rootJson))
```

Comment:
Should we use `Utf8JsonReader` here instead of `JsonDocument`?

Reply:
`JsonDocument` is indeed much simpler to use than `Utf8JsonReader`. Since `JsonDocument` already uses `Utf8JsonReader` in its implementation, why don't we use `JsonDocument` everywhere?

Reply:
JsonDocument loads and completely parses an entire JSON object. Even properties that one might not care about. 

Reader allows precise reading, skipping is available for tokens that are unnecessary. 

For feature filters parameters, their type is unknown. In this case, JsonDocument is the correct approach. We gain no benefits from reader.

Reply:
Makes sense, thanks.